docs/proposals/resource-quota-reservations.md

```diff
@@ -0,0 +1,241 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Resource Quota - reservations
+
+## Problem Description
+
+When a request for creating a resource is received,
+the associated resource quota is checked by the
+server in admission control.  If the request would
+not violate the quota, the quota usage is incremented
+and the requesting object is allowed to be created.
+Quota usage is incremented using compare-and-swap
+to allow for optimistic locking.  If the quota usage
+was stale, the admission control logic runs again
+to validate the request is still valid with the latest
+quota document.
+
+A quota controller regularly sweeps the system to
```

Comment:
@lavalamp - deletions return quota for pods today immediately.  I have code locally that I am opening shortly to get the same behavior for the other resource types.